src/pages/sidepanel/index.tsx

```diff
@@ -11,8 +10,69 @@ function init() {
   if (!appContainer) {
     throw new Error('Can not find #app-container');
   }
-  const root = createRoot(appContainer);
-  root.render(<SidePanel />);
+  ReactDOM.render(<IndexSidepanel />, appContainer);
 }
 
 init();
+
+// This code sends a message to the background script with a greeting.
+// Upon receiving the message, the background script sends a response with a farewell message.
+
+chrome.runtime.sendMessage({ greeting: 'Hello' }, function (response) {
+  console.log(response.farewell);
+});
+
+chrome.runtime.onMessage.addListener(
+  function(request, sender, sendResponse) {
+    console.log(sender.tab? "from a content script:" + sender.tab.url : "from the extension");
+    if (request.greeting === "hello") sendResponse({farewell: "goodbye"});
+  });
+
+function IndexSidepanel(): JSX.Element {
+  const [currentUrl, setCurrentUrl] = useState<string>("");
+  const [formData, setFormData] = useState<{ name: string, email: string }>({ name: '', email: '' });
+
+  const getCurrentUrl = async (): Promise<void> => {
+    const [tab] = await chrome.tabs.query({ active: true, currentWindow: true });
+    setCurrentUrl(tab.url || 'unknown');
+  };
+
+  useEffect(() => {
+    getCurrentUrl();
+  }, [currentUrl]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData(prevState => ({...prevState, [name]: value }));
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+      if (tabs[0]?.id) {
+        chrome.tabs.sendMessage(tabs[0].id, { action: "fillForm", data: formData });
+      }
+    });
+  };
+
+  return (
+    <div>
+      <h1>You are currently at {currentUrl}</h1>
```

Comment:
As I said before we should not have static formData fields like name, email . All should be dynamic.